packages/itwin/property-grid/src/components/PropertyGrid.tsx

```diff
@@ -200,19 +258,7 @@ export const PropertyGrid = ({
     },
 
     [iModelConnection, favoritePropertiesScope]
-  );
-
-  const onHideNull = useCallback(() => {
-    setFilterer(new NonEmptyValuesPropertyDataFilterer());
```

Comment:
why is the NonEmptyValuesPropertyDataFilterer no longer used?

Reply:
As I mentioned in the PR description, I updated it to use this property from the data provider itself: `AutoExpandingPropertyDataProvider.includeFieldsWithNoValues`. The filter was only used for this purpose and seemed like overkill when there's a built-in way to handle it already.

Reply:
was it not working correctly? why do this now?

Reply:
I mean I can switch it back if you really want, it just seemed like an overly complicated way to do it when there's already an option in the data provider and nothing else in property grid uses the filter. And if there's a simple way to do it, then when what better time to change than when I'm already updating the code?

Reply:
couple thoughts...

- if you make this change then you have to consider the following
   - If you are removing the usage of NonEmptyValuesPropertyDataFilterer should you go ahead an also remove the PlaceholderPropertyDataFilterer (does that one only exist to counter the NonEmptyValues one and have no purpose when by itelf)
   - if you are removing these and they are no longer used should you remove the definitions for these filters from the source code
   - are you positive this change introduces no regression

wrt to 'what better time to change than when I'm already updating the code', i think that the scope of a PR is important. That is to say that just because you're modifying a file does not mean it's prudent to modify anything in that file worth changing. Here a couple reasons why:
- those changes may get conflated with the purpose of the PR they were made in. i.e. a developer may assume these changes explicitly relate to persisting null values versus just a more efficient way to toggle null values
- code change has inherent risk and if you include changes in a PR that are unnecessary (strictly speaking), if they cause an issue they will impede the resolution of the issue the PR intends to fix whereas if those changes were isolated they could also be resolved separately
- changes unrelated to the express purpose of a PR will obfuscate the changes that are relevant and make it more challenging to identify them for improvements/fixes later on (this is also why moving code more than a few lines into a separate file that has a single responsibility is a good thing to do as it makes it quicker and less risky to modify code pertaining to that responsibility)